src/components/GlobalPreferences/CustomLabels/ShareModal.js

```diff
@@ -0,0 +1,197 @@
+              width: 100%;
+              position: relative;
+              background: ${theme.surface};
+              border: 1px solid ${theme.border};
+              border-radius: 3px;
+              box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.06);
+              padding-right: 30px;
+              margin-top: ${1 * GU}px;
+            `}
+          >
+            <TextInput
+              ref={inputRef}
+              value={link}
+              onFocus={onFocus}
+              readOnly
+              css={`
+                text-overflow: ellipsis;
+                width: 100%;
+                max-width: 100%;
+                border: 0;
+                box-shadow: none;
+                background: ${theme.surface};
+                &:read-only {
+                  color: ${theme.content};
+                  text-shadow: none;
+                }
+              `}
+            />
+            <ButtonIcon
+              onClick={onCopy}
+              label="Copy to clipboard"
+              css={`
+                position: absolute;
+                top: 0;
+                right: 0;
+                width: 39px;
+                height: 38px;
+                display: flex;
+                align-items: center;
+                justify-content: center;
+                border-radius: 0 3px 3px 0;
+                &:active {
+                  background: rgba(220, 234, 239, 0.3);
+                }
+              `}
+            >
+              <IconCopy />
+            </ButtonIcon>
+          </div>
+        </div>
+      </main>
+      <footer
+        css={`
+          margin-top: ${3 * GU}px;
+          display: flex;
+          justify-content: space-between;
+          text-align: right;
+
+          ${breakpoint(
+            'medium',
+            `
+              display: block;
+            `
+          )}
+        `}
+      >
+        <Button label="Close modal" mode="secondary" onClick={onClose}>
+          Close
+        </Button>
+        <Button
+          mode="strong"
+          label="Copy link to clipboard"
+          onClick={onCopy}
+          css={`
+            margin-left: ${2 * GU}px;
+          `}
+        >
+          Copy link
+        </Button>
+      </footer>
+    </Modal>
+  )
+}
+
+ShareModal.propTypes = {
+  inputRef: PropTypes.object,
+  onClose: PropTypes.func.isRequired,
+  onCopy: PropTypes.func.isRequired,
+  onFocus: PropTypes.func.isRequired,
+  link: PropTypes.string.isRequired,
+  visible: PropTypes.bool.isRequired,
+}
+
+function ShareModalWithToast({ toast, visible, onClose, link }) {
+  const inputRef = useRef()
+  const handleFocus = useCallback(() => {
+    inputRef.current.select()
+  }, [inputRef])
+  const handleCopy = useCallback(() => {
+    inputRef.current.focus()
+    inputRef.current.select()
+    try {
+      document.execCommand('copy')
+      toast('Link copied')
+      setTimeout(onClose, (TIMEOUT_TOAST * 7) / 8)
+    } catch (err) {
+      console.warn(err)
```

Comment:
I don't think this will ever happen but if it does and we get a report about it the log might tell us what happened, again, I think it is highly unlikely to occur.

Reply:
Even if super unlikely, I would find it more useful to inform the user than silently failing. But if it can really never fail, then we don’t need the warn :smile:

Reply:
Do we really need a 3 day back and forth for a `toast` vs `console.warn` concern? I do believe this could be improved if and when we find a case for it but shouldn't we focus our efforts on prioritizing critical tasks?
cc @luisivan

Reply:
We don’t. It’s not really about a toast vs. console warning on this specific line, but more about sharing ideas in context about how we should handle these situations.